AfterClass-io/afterclass.io-v2
# feat: implement like button functionality on review item

closes #240 

## Changes

- like button is now reactive and functional

## Implementation Details



- only logged in users can like
- add review votes trpc router with 3 procedures:
    - `count`: num of votes to be displayed on the button
    - `hasUserVoted`: to check if user has voted to highlight button if voted
    - `voteOrUnvote`: to add or remove vote of the user
- abstract button logic from reviewergroup component
- overall logic:
    - on click, if user has already liked, then unvote the review. otherwise, vote the review highlight the button
    - after above mutation, refetch review vote count as it is cached

## How to Test



1. head to preview link
2. click a like
3. see nothing happens, expected as only authenticated users can like
4. login
5. click a like
6. see button highlights and count increase by 1
7. refresh
8. see button highlighted and count is the old count + 1
9. click the already liked button
10. see button dulls and count decrease by 1
11. refresh
12. see button dull and count is the old count - 1

## Preview / Screenshots




![image](https://github.com/user-attachments/assets/462a765a-618f-485f-8ff9-843ee066561d)

![image](https://github.com/user-attachments/assets/f2f548f0-283a-404b-a460-06f18ea07d90)

## Checklist
